src/trie/merkle_tree.rs

```diff
@@ -177,7 +177,29 @@ impl<H: StarkHash + Send + Sync, DB: BonsaiDatabase, CommitID: Id> MerkleTrees<H
                 // Remove the identifier from the key
                 key_value_pairs
                     .into_iter()
-                    .map(|(key, _value)| key[identifier.len()..].to_vec())
+                    // FIXME: this does not filter out keys values correctly for `HashMapDb` due
+                    // to branches and leafs not being differenciated
+                    .filter(|(key, _value)| key.len() > identifier.len())
```

Comment:
Can you add a comment explaining under which condition this may not happen?

Reply:
This should always happen for any `HashMapDb` with node count > 1. More details have been provided in #26.